src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java

```diff
@@ -113,11 +114,26 @@ public ActiveDirectorDetails getActiveDirectorDetails(String companyNumber) thro
         ResponseEntity<ShareholderJson[]> response = restTemplate.getForEntity(shareholdersUrl, ShareholderJson[].class);
         LOGGER.info(RECEIVED_FROM_ORACLE_QUERY_API_URL_GET, response, shareholdersUrl);
         if (response.getStatusCode() != HttpStatus.OK) {
-            throw new ServiceException(String.format("Oracle query api returned with status = %s, companyNumber = %s", response.getStatusCode(), companyNumber));
+            throw new ServiceException(String.format(ORACLE_QUERY_API_STATUS_MESSAGE, response.getStatusCode(), companyNumber));
         }
         if (response.getBody() == null) {
             return new ArrayList<>();
         }
         return Arrays.asList(response.getBody());
     }
+
+    public boolean isConfirmationStatementPaid(String companyNumber, String dueDate) throws ServiceException {
+       var paymentsUrl = String.format(
+               "%s/company/%s/confirmation-statement/paid/?payment_period_due_date=%s", oracleQueryApiUrl, companyNumber, dueDate);
+
+        LOGGER.info(CALLING_ORACLE_QUERY_API_URL_GET, paymentsUrl);
+        ResponseEntity<Boolean> response = restTemplate.getForEntity(paymentsUrl, Boolean.class);
+        if (response.getStatusCode() != HttpStatus.OK) {
+            throw new ServiceException(String.format(ORACLE_QUERY_API_STATUS_MESSAGE, response.getStatusCode(), companyNumber));
+        }
+        if (response.getBody() == null) {
+            throw new ServiceException("Oracle query api returned null, boolean values expected");
```

Comment:
I'd also add the due date into the 2 service exception messages, it'll help us to identify the payment periods when these error are thrown in Live.